Feature/date storage and display

HassanOHOsman opened the pull request:
This PR implements storing and displaying user-specific agenda items. Users can add a topic and date, which are saved without overwriting existing entries. Selecting a user from the dropdown displays all saved agenda items for that user, including newly added ones.

HassanOHOsman committed:
feat: store new data and display full agenda

HassanOHOsman committed:
feat: data storage and display for users

sheida-shab reviewed (commented):
The calculateRevisionDate function is already there and working, it’s just not being used in the data-saving part yet.
You just need to call the calculateRevisionDate function inside the submit handler and use a loop to prepare and save the data for each future date. This way, both the original date and all future revision dates will be stored properly.

sheida-shab commented:
Just to be sure, I wanted to make sure you also pay attention to these points when displaying the information: 

If there is an agenda, your website should display a list of topics and the revision date when the user should revise in _chronological order_. For each revision date, it should display the date and the name of the topic to revise.   ****_Revision dates in the past should not be displayed._****

Your website must include a form with a text input, a date picker and submit button that allows a user to add a new topic. This form must be accessible, so for example,   ****_hitting the Enter key will also submit the topic name, the same as clicking the submit button._****

HassanOHOsman committed:
feat: update sort &data storage &fix date function

Bluejay600 committed:
I have Updated the following part of form.addEventListener("submit", ...) code to:

Call calculateRevisionDate() with the selected date.
Store each revision date as a separate entry in the agenda.
Add those dates using addData().
Modify showUserAgenda() to sort the agenda by date before rendering.

Bluejay600 committed:
Modifyied showUserAgenda() to sort the agenda by date before rendering.

Bluejay600 committed:
Merge remote-tracking branch 'refs/remotes/origin/feature/date-storage-and-display' into feature/date-storage-and-display

sheida-shab reviewed (commented):
Well done Hassan.All changes look good.Just a few changes  about not showing in user agenda the past dates.
I tested with this instruction that provided in readme and the result was not as it shoud be.

Steps:

1. Select User 3 from the drop-down
1. Add “Codewars” to the text input
1. Select the date exactly one month ago (e.g. if it's currently July 26th, select June 26th this year) from the date picker
1. Submit the form

Expected result:

- The agenda for user 3 is shown, with the revision dates shown as follows:
  - (No topic is shown for 1 week after the selected date, as this is in the past)
  - Codewars, Today's date (e.g. if it's currently July 26th, shows: July 26th)
  - Codewars, Two months in the future (e.g. if it's currently July 26th, shows: September 26th)
  - Codewars, 5 months in the future (e.g. if it's currently July 26th, shows: December 26th)
  - Codewars, 11 months in the future, (e.g. if it's currently July 26th, shows: June 26th)
- Each of the revision dates show the topic name and the relevant date (styling/formatting does not matter as long as it is understandable)
- The form remains on the website (allowing for further topics to be added)
